File: frontend/src/api/schedule.ts
import http from './index'

export interface Schedule {
  id?: number
  courseId: number
  teacherId: number
  classroom: string
  dayOfWeek: number
  startTime: string
  endTime: string
  capacity?: number
  courseName?: string
  teacherName?: string
  createdAt?: string
  updatedAt?: string
}

export interface ScheduleDTO {
  courseId: number
  teacherId: number
  classroom: string
  dayOfWeek: number
  startTime: string
  endTime: string
  capacity?: number
}

export function listSchedules(params: {
  courseId?: number; teacherId?: number; classroom?: string; dayOfWeek?: number; page?: number; size?: number
}) {
  return http.get('/schedules', { params }).then(r => r.data)
}

export function getSchedule(id: number) {
  return http.get(`/schedules/${id}`).then(r => r.data)
}

export function createSchedule(data: ScheduleDTO) {
  return http.post('/schedules', data).then(r => r.data)
}

export function updateSchedule(id: number, data: ScheduleDTO) {
  return http.put(`/schedules/${id}`, data).then(r => r.data)
}

export function deleteSchedule(id: number) {
  return http.delete(`/schedules/${id}`).then(r => r.data)
}

File: backend/src/main/java/com/sims/entity/Schedule.java
package com.sims.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Data
@TableName("course_schedule")
public class Schedule {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long courseId;
    private Long teacherId;
    private String classroom;
    private Integer dayOfWeek;
    private LocalTime startTime;
    private LocalTime endTime;
    private Integer capacity;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableField(exist = false)
    private String courseName;

    @TableField(exist = false)
    private String teacherName;
}
